src/Microsoft.TemplateEngine.Edge/TemplateUpdates/InstallUnitDescriptorFactory.cs

```diff
@@ -10,8 +10,23 @@ namespace Microsoft.TemplateEngine.Edge.TemplateUpdates
 {
     public static class InstallUnitDescriptorFactory
     {
-        public static bool TryParse(IEngineEnvironmentSettings environmentSettings, JObject descriptorObj, out IInstallUnitDescriptor parsedDescriptor)
+        public static bool TryParse(IEngineEnvironmentSettings environmentSettings, JProperty descriptorProperty, out IInstallUnitDescriptor parsedDescriptor)
         {
+            if (descriptorProperty == null)
+            {
+                parsedDescriptor = null;
+                return false;
+            }
+
+            if (!Guid.TryParse(descriptorProperty.Name, out Guid descriptorId)
+                || descriptorId == Guid.Empty)
+            {
+                parsedDescriptor = null;
+                return false;
+            }
+
+            JObject descriptorObj = descriptorProperty.Value as JObject;
+
             if (descriptorObj == null)
```

Comment:
Seems like we'd be able to do `if (!(descriptorProperty.Value is JObject descriptorObj))` here